src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt

```diff
@@ -321,30 +322,142 @@ class CreateMafVcf : CliktCommand(help = "Create gVCF and hVCF from Anchorwave M
             else lastVariant.getAttributeAsInt("ASM_Start",region.second.position)
         }
 
+        val regions = buildNewAssemblyRegions(newASMStart,newASMEnd,variants)
+
 
         return HVCFRecordMetadata(sampleName=sampleName, refSeq = refRangeSeq.toString(), asmSeq = "",
             refContig = region.first.contig, refStart = region.first.position, refEnd = region.second.position,
-            asmContig = firstVariant.getAttributeAsString("ASM_Chr",""),
-            asmStart = newASMStart, asmEnd = newASMEnd, asmStrand = strand
-        )
+            regions)
+
+    }
+
+    /**
+     * Function to build the new assembly region coordinates based on the new Start and end and the list of VariantContexts
+     * Any consecutive regions should be merged together so we do not make the eventual string too long
+     * The output will be a List<Pair<Position,Position>> which will be the new coordinates for the all the  assembly regions
+     */
+    fun buildNewAssemblyRegions(newStart: Int, newEnd: Int, variants: List<VariantContext>) : List<Pair<Position,Position>> {
+        val variantsConverted = variants.map { convertVariantContextToPositionRange(it) }
+
+        //resize the first and last position based on the strand
+        val resizedFirst = resizePositionRange(variantsConverted.first(),newStart,true)
+        val resizedLast = resizePositionRange(variantsConverted.last(),newEnd,false)
+
+        //merge the first and last with the rest of the variants
+        val mergedVariants = mutableListOf<Pair<Position,Position>>()
+        mergedVariants.add(resizedFirst)
+        mergedVariants.addAll(variantsConverted.subList(1,variantsConverted.size-1))
```

Comment:
Shouldn't this be variantsConverted.subList(0,variantsConverted.size-1)?  When the list only has 1 element, you get an out of range exception as the range goes from 1 to 0.